From 14bdf1487a0bfdc9c1f31c2387aa6e4eb71d7bbb Mon Sep 17 00:00:00 2001
From: Ramanathan Choodamani <quic_rchoodam@quicinc.com>
Date: Fri, 17 Feb 2023 01:50:42 -0800
Subject: [PATCH 2/7] ath12k: Update iowrite32 to
 write_relaxed to boost performance

Use writel_relaxed instead iowrite32 to avoid the memory barrier
instuction and slowdown the cpu execution performance.
This helps improve the throughput by 150Mbps

Signed-off-by: Balamurugan Mahalingam <quic_bmahalin@quicinc.com>
Signed-off-by: Ramanathan Choodamani <quic_rchoodam@quicinc.com>
---
 drivers/net/wireless/ath/ath12k/hal.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/net/wireless/ath/ath12k/hal.h b/drivers/net/wireless/ath/ath12k/hal.h
index a195115..c75b66a 100644
--- a/drivers/net/wireless/ath/ath12k/hal.h
+++ b/drivers/net/wireless/ath/ath12k/hal.h
@@ -1245,7 +1245,7 @@ void ath12k_hal_srng_access_umac_src_ring_end_nolock(void *mem,
 
 	srng->u.src_ring.last_tp =
 			*(volatile u32 *)srng->u.src_ring.tp_addr;
-	iowrite32(srng->u.src_ring.hp, mem + HAL_DP_REG_WINDOW_OFFSET
+	writel_relaxed(srng->u.src_ring.hp, mem + HAL_DP_REG_WINDOW_OFFSET
 		  + (offset & WINDOW_RANGE_MASK));
 	srng->timestamp = jiffies;
 }
-- 
2.7.4

